src/serializer.js

```diff
@@ -223,13 +179,33 @@ const MapHandler = [
   }),
 ];
 
-export const writer = (customWriters = [], opts = {}) => {
-  const ownHandlers = constructWriteHandlers(customWriters);
+export const writer = (appTypeWriters = [], opts = {}) => {
   const { verbose } = opts;
   const transitType = verbose ? 'json-verbose' : 'json';
 
   return transit.writer(transitType, {
-    handlers: transit.map([...ownHandlers, ...MapHandler]),
+    handlers: transit.map([...writeHandlers, ...MapHandler]),
+
+    // Use transform to transform app types to sdk types before sdk
+    // types are encoded by transit.
+    transform: v => {
+      if (v && v instanceof Object) {
```

Comment:
What are you after with `v instanceof Object`? Is it really needed, or should a Lodash helper like [_.isObject](https://github.com/lodash/lodash/blob/4.17.11/lodash.js#L11744) be used?

Reply:
@kpuputti Good question.

What I was after here was that the SDK will make some checks that make the `canHandle` function implementations easier and less error-prone. By this I mean that the `canHandle` function implementation can safely do stuff like `v => v.isThisMyUuidType`. This will obviously fail is `v` is `null` or `undefined`, thus the `null` check is done by SDK. I also *assumed* that this would fail for primitive types like numbers and boolean, thus I added the `instanceof Object` check. However, now that I tested it, it seems that the `instanceof` check may be useless, because these seem to work just fine:

```
const a = 1;
a.isThisMyType ;; => undefind
const b = false;
b.isThisMyType ;; => undefind
```

So maybe I'll remove the `instanceof` check.